pilot/pkg/networking/core/v1alpha3/envoyfilter/rc_patch.go

```diff
@@ -249,9 +251,6 @@ func patchHTTPRoute(patchContext networking.EnvoyFilter_PatchContext,
 			routeConfigurationMatch(patchContext, routeConfiguration, rp, portMap) &&
 			virtualHostMatch(virtualHost, rp) &&
 			routeMatch(virtualHost.Routes[routeIndex], rp) {
-
-			// different virtualHosts may share same routes pointer
-			virtualHost.Routes = cloneVhostRoutes(virtualHost.Routes)
```

Comment:
maybe instead of full clone we just shallow copy the list, and then when we merge (L260) we just copy+merge that single route?

Even with your patch we still clone more than we need in many cases?

Reply:
> maybe instead of full clone we just shallow copy the list, and then when we merge (L260) we just copy+merge that single route?
> 
> Even with your patch we still clone more than we need in many cases?

You mean we need to identify which router pointers are shared by different virtualHosts, we just do a deep copy of the shared routers so that we won't have clones we don't need, but this doesn't seem easy to implement

Reply:
Not even identify them, just on L260 instead of `proto.Clone(original)` we do `proto.Clone(copy(original))`. Then we copy a single one instead of the whole list.

Imagine we have 1000 routes and just modify one

Reply:
> Not even identify them, just on L260 instead of `proto.Clone(original)` we do `proto.Clone(copy(original))`. Then we copy a single one instead of the whole list.
> 
> Imagine we have 1000 routes and just modify one

I have submitted new code, you can review the new code